src/trie.c

```diff
@@ -0,0 +1,92 @@
+/*
+   A trie data structure
+*/
+
+#include <stdlib.h>
+#include <assert.h>
+#include <math.h>
+#include "trie.h"
+#include "utils.h"
+
+/* See trie.h */
+trie_t *trie_new(char current) {
+    trie_t *t = calloc(1,sizeof(trie_t));
+
+    if (t == NULL) {
+        error("Could not allocate memory for trie_t");
+        return NULL;
+    }
+
+    t->current = current;
+
+    t->children = calloc(256,sizeof(trie_t*));
+    if (t->children == NULL) {
+        error("Could not allocate memory for t->children");
+        return NULL;
+    }
+
+    t->is_word = 0;
+    t->parent = NULL;
+
+    return t;
+}
+
+/* See trie.h */
+int trie_free(trie_t *t) {
+    assert( t != NULL);
+
+    for (int i=0; i<256; i++ ) {
```

Comment:
The styling in this function is still really ugly. Please pass all files through `astyle`.

Reply:
Addressed